src/js/contexts/AnalyticsContext/AnalyticsContext.js

```diff
@@ -0,0 +1,40 @@
+import React, { useCallback, useContext, useEffect, useRef } from 'react';
+
+export const AnalyticsContext = React.createContext(() => {});
+
+export const useAnalytics = () => useContext(AnalyticsContext);
+
+export const AnalyticsProvider = ({ onAnalytics, children }) => {
+  const pathInfoRef = useRef({
+    lastUrl: window.location.href,
```

Comment:
In SSR app, window will not be available yet. Can we initialize as undefined?

```suggestion
    lastUrl: undefined,
```